### PR TITLE
Expose `Geometry2D.bresenham_line()` method

### PR DESCRIPTION
**Pull request by dalexeev**
This method can be useful in user code.

* Closes https://github.com/godotengine/godot-proposals/discussions/11004.

### CODE REVIEW

**Comment by MewPurPur** [comment]
You could also update the InputEventMouseMotion description, which suggests "If you use InputEventMouseMotion to draw lines, consider implementing [url=https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm]Bresenham's line algorithm[/url] as well to avoid visible gaps in lines if the user is moving the mouse quickly."

**Review by timothyqiu — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

File: doc/classes/Geometry2D.xml [hunk 1]
```diff
@@ -9,6 +9,14 @@
 	<tutorials>
 	</tutorials>
 	<methods>
+		<method name="bresenham_line">
+			<return type="Vector2i[]" />
+			<param index="0" name="from" type="Vector2i" />
+			<param index="1" name="to" type="Vector2i" />
+			<description>
+				Returns the [url=https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm]Bresenham line[/url] between the [param from] and [param to] points.
```
Comment (Calinou) [inline doc/classes/Geometry2D.xml:17]:
```suggestion
				Returns the [url=https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm]Bresenham line[/url] between the [param from] and [param to] points. A Bresenham line is a series of pixels that draws a line and is always 1-pixel thick on every row and column of the drawing (never more, never less).
				Example code to draw a line between two [Marker2D] nodes using a series of [method CanvasItem.draw_rect] calls:
				[codeblock]
				func _draw():
				    for pixel in Geometry2D.bresenham_line($MarkerA.position, $MarkerB.position):
				        draw_rect(Rect2(pixel, Vector2i(1, 1)), Color.WHITE)
				[/codeblock]
```

Comment (Calinou) [inline doc/classes/Geometry2D.xml:17]:
```suggestion
				Returns the [url=https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm]Bresenham line[/url] between the [param from] and [param to] points. A Bresenham line is a series of pixels that draws a line and is always 1-pixel thick on every row and column of the drawing (never more, never less).
				Example code to draw a line between two [Marker2D] nodes using a series of [method CanvasItem.draw_rect] calls:
				[codeblock]
				func _draw():
				    for pixel in Geometry2D.bresenham_line($MarkerA.position, $MarkerB.position):
				        draw_rect(Rect2(pixel, Vector2i(1, 1)), Color.WHITE)
				[/codeblock]
```

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected. Should be good to merge after applying the suggestion.

**Testing project:** [test_pr_74714.zip](https://github.com/user-attachments/files/17513322/test_pr_74714.zip)

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**